### PR TITLE
feat(Reactions): add video frames

### PR DESCRIPTION
**Pull request by OCannings**
A new VideoFrame component has been added which masks a video and
renders a decorative frame on top of it.

```
<VideoFrame src="hello.mp4" frame="frame-1" />
```

The videos autoplay and loop.

### CODE REVIEW

**Comment by ljkbennett** [comment]
:eyes:

**Comment by OCannings** [comment]
still WIP - opening for comments

File: apps/juicebox_web/assets/js/reactions/video-frame.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React, { Component, PropTypes } from 'react';
+const frameAsset = require.context('../../images/frames', false);
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/video-frame.js]: controversial

Comment (OCannings) [reply apps/juicebox_web/assets/js/reactions/video-frame.js]: :D suggestions welcome

Comment (ljkbennett) [reply apps/juicebox_web/assets/js/reactions/video-frame.js]: No idea really. Would `import * as frameAssets from '../../images/frames';` not work?

Comment (OCannings) [reply apps/juicebox_web/assets/js/reactions/video-frame.js]: I don't think so, but I'll revisit this before merging.

Comment (ljkbennett) [reply apps/juicebox_web/assets/js/reactions/video-frame.js]: It's no more standards-compliant than this anyway. I think the only way to do it that would be would be to have a manifest file where you export the frames.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 6 possible new issues (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 fixed issues! 🎉

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

**Comment by ljkbennett** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

File: apps/juicebox_web/assets/js/reactions/player.js [hunk 2]
```diff
@@ -1,33 +1,18 @@
...
     return (
-      <video
-        className={ this.state.isNew ? 'new-reaction' : '' }
-        autoPlay
-        ref={ (video) => { this.video = video } }
+      <VideoFrame
         src={ this.props.reaction }
-        onEnded={ this.onLoop }>
-      </video>
+        frame="frame-1"
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/player.js]: hard coded?

File: apps/juicebox_web/assets/js/reactions/reactions.js [hunk 3]
```diff
@@ -3,19 +3,33 @@ import { connect } from 'react-redux';
 import { map } from 'lodash';
 import { sendReaction } from './actions';
 import ReactionRecorder from './recorder';
-import ReactionPlayer from './player';
+import VideoFrame from './video-frame';
+import FrameSelector from './frame-selector';
+import { FRAMES } from './frames';
+import './styles.scss';
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/reactions.js:9]: I've switched to modular styles so I may need to fix this.

**Review by ljkbennett — APPROVED** [review]

File: apps/juicebox_web/assets/js/reactions/reactions.js (hunk 3 above)
Comment (OCannings) [reply apps/juicebox_web/assets/js/reactions/reactions.js:9]: That shouldn't be a problem, there isn't much in there.

**Pull request merged**